### ISSUE TITLE
Как встроить AWOOC после надписи Купить (в один ряд)?

### ISSUE THREAD

**Issue by Antarix76** [artikus11/art-woocommerce-order-one-click#54]
Доброго времени суток... Несколько дней бьюсь над проблемой... Не могу никак в каталоге товаров вставить [AWOOC] в одни ряд с кнопкой, а точнее ссылкой Купить... На странице товара нет проблем, там кнопка купить есть button а на странице каталога ссылка внутри Li и AWOOC  встраивается за /div> перед /li>

![Screenshot_1](https://user-images.githubusercontent.com/127436257/224123802-81bdfcbf-4fc1-45dd-9093-d7b314d94f9b.jpg)
------------------------------------------------------------------------------
![Screenshot_2](https://user-images.githubusercontent.com/127436257/224123808-fd666210-f658-4031-8745-6d711f871f4d.jpg)
-------------------------------------------------------------------------------
![Screenshot_3](https://user-images.githubusercontent.com/127436257/224123809-01c6df3b-f9f2-4bde-87d9-6cb5db42e59b.jpg)
-----------------------------------------------------------------------------
![Screenshot_4](https://user-images.githubusercontent.com/127436257/224123811-bbe9265a-8ea9-4f98-8dfe-c4a2598967ba.jpg)

**Comment by artikus11** [comment]
@Antarix76 Доброго! Такое может быть, если  в теме переделан вывод карточек. Надо хук другой использовать если он вообще есть или напрямую как то функцию выводить. Что за тема у вас?

**Comment by Antarix76** [comment]
> @Antarix76 Доброго! Такое может быть, если в теме переделан вывод карточек. Надо хук другой использовать если он вообще есть или напрямую как то функцию выводить. Что за тема у вас?

Тема Астра... Я ваш плагин уже ставил и проблем не было... Я не большей спец в коде, но не могу найти как его запихнуть над закрытие дива... 


![Screenshot_5](https://user-images.githubusercontent.com/127436257/224124770-bdf27c07-77d9-4ffa-875f-62a702cb9077.jpg)

**Comment by artikus11** [comment]
@Antarix76 вот так можно на других хуках
```
//Обертка кнопок
add_filter( 'astra_woo_shop_add_to_cart_before', function () {

	?>
	<div class="wrap-buttons" style="display: flex; gap: 10px;">
	<?php
} );

// Вывод функции и закрытие обертки
add_filter( 'astra_woo_shop_add_to_cart_after', function (){
	if (is_product()){
		return;
	}

	awooc_html_custom_add_to_cart();
	?>
	</div>
	<?php
} );
```

**Comment by Antarix76** [comment]
> @Antarix76 вот так можно на других хуках

Спасибо огромное, работает, сейчас проверю везде... СПАСИБО

![Screenshot_6](https://user-images.githubusercontent.com/127436257/224127835-0af3ca52-76b3-4931-85db-656fb33af4d4.jpg)

**Comment by Antarix76** [comment]
Есть правда ещё одна проблема в каталоге с вариативным товаром... В форму не передаётся выбранный вариант... В сингл продукт такого нет...

![Screenshot_7](https://user-images.githubusercontent.com/127436257/224130028-d7df88a7-04a7-4230-8c75-202043052bc6.jpg)
![Screenshot_8](https://user-images.githubusercontent.com/127436257/224130040-675bb6fe-46fa-4abc-9905-3b0422bddd41.jpg)

**Comment by artikus11** [comment]
@Antarix76 

> Есть правда ещё одна проблема в каталоге с вариативным товаром... В форму не передаётся выбранный вариант... В сингл продукт такого нет...
Это уже сложно, надо получать что выбрано на фронте  передавать в форму. Сложна, так с наскока не подскажу

**Comment by Antarix76** [comment]
> Это уже сложно, надо получать что выбрано на фронте передавать в форму. Сложна, так с наскока не подскажу

Да может с наскока и не обязательно, я подожду, если у вас время будет, буду весьма благодарен.... А то с вариативным получается нет смысла от кнопки :-)

**Comment by artikus11** [comment]
Ну универсальное решение тут не получиться сделать, так как по дефолту выбора вариации в карточке на каталоге нет, а значит тут или к конкретному проекту привязываться или к плагину, который выводить в карточке вариации
Как у вас сделан этот вывод вариаций?

**Comment by Antarix76** [comment]
> Ну универсальное решение тут не получиться сделать, так как по дефолту выбора вариации в карточке на каталоге нет, а значит тут или к конкретному проекту привязываться или к плагину, который выводить в карточке вариации Как у вас сделан этот вывод вариаций?

С плагином, остановился на этом...
https://ru.wordpress.org/plugins/variation-swatches-woo/

**Comment by artikus11** [comment]
> С плагином, остановился на этом...
https://ru.wordpress.org/plugins/variation-swatches-woo/

Он в бесплатной версии умеет в каталоге вариации выводить?

**Comment by Antarix76** [comment]
> А вижу, вроде умеет. Ну ок, на выходных попробую посмотреть, подумать как передавать из него данные
Да, умеет, и это не первый, что я пробовал был ещё один, но с ним что-то было не так и я остановился на этом...

> Ну ок, на выходных попробую посмотреть,
Огромное спасибо...

**Comment by artikus11** [comment]
@Antarix76 Сделал поддержку с этим плагином, на тестовом у меня работает и атрибуты подтягиваются. Проверьте у себя. ТОлько новую версию скачайте

**Comment by artikus11** [comment]
>  отключил хук добавленный ранее...

Хм, что за хук? Кнопку то надо выводить в любом случае

**Comment by Antarix76** [comment]
> > отключил хук добавленный ранее...
> 
> Хм, что за хук? Кнопку то надо выводить в любом случае

Я с этим погорячился, вернул кнопку назад...  Тут вот какая штука... в карточке товара выбор начинает работать, когда я несколько раз нажимаю по всем вариантам выбора товара и в какой-то момент всё начинает работать, и в форме заказа всё правильно выбирается... в Категории делаю всё тоже самое и там начинает выбираться, но выбор в форму заказа не передаётся...

**Comment by Antarix76** [comment]
Одно я вроде понял, в настройке вариативного товара есть опция "Значения форм по умолчанию:"  у меня там три варианта и из трёх по умолчанию я сделал только 2, чтоб была кнопка Выбрать... надо выбрать цвет, тогда появляется цена, а сейчас такое ощущение что все формы "по умолчанию" сняты и получается что надо кликнуть по всем трём вариантам и только тогда появляется цена и возможность купить...

**Comment by artikus11** [comment]
Ну естественно, если в товаре 3 вариации то пока комбинация вариаций не выбрана, купить товар не получиться

**Comment by Antarix76** [comment]
> Ну естественно, если в товаре 3 вариации то пока комбинация вариаций не выбрана, купить товар не получиться

Да это я понимаю, поэтому по умолчанию 2 уже были выбраны а третий цвет надо было выбрать самому... А сейчас все три надо выбрать... Это по сути не есть проблема, но такое... А вот в каталоге товара у меня почемуто не выбирается... не меняются в форме заказа варианты
![Screenshot_2](https://user-images.githubusercontent.com/127436257/224691722-8dd57c92-5533-4f8b-93ad-f7b5a03d0d5e.png)
![Screenshot_3](https://user-images.githubusercontent.com/127436257/224691725-c89b18e2-a07d-4722-a454-434cffa2d9c0.png)

**Comment by artikus11** [comment]
Это уже смотреть надо, могут быть и глюки, дайте ссылку на сайт, посмотрю в выходные

**Comment by Antarix76** [comment]
> Это уже смотреть надо, могут быть и глюки, дайте ссылку на сайт, посмотрю в выходные

Отправил в Телеграм

**Comment by artikus11** [comment]
Ага, вижу

**Comment by Antarix76** [comment]
> > Это уже смотреть надо, могут быть и глюки, дайте ссылку на сайт, посмотрю в выходные
> 
> Отправил в Телеграм
На всякий случай напоминаю за выходные...
